mexico-city/src/managers/chihuahua_manager.rs

```diff
@@ -210,7 +214,7 @@ fn dispatch_on_program(
             // program provisioner can just try again.
             assert!(check_state(
                 &protocol_state.get_lifecycle_state()?,
-                &[LifecycleState::Initial]
+                &[LifecycleState::Error, LifecycleState::Initial]
```

Comment:
Is there a description somewhere of what the new state transition diagram is?  Maybe we need to update the contents of e.g. the Wiki to keep track of what states can transition to other states, etc.

Reply:
For this particular change, it is a bug fixed. In the `load_program` function, if it fails, it will transfer to Error state. 
Yes, I will document down somewhere about the new state machine.